### PR TITLE
Unity builder v4 Parity Update

### PR DESCRIPTION
**Pull request by AndrewKahr**
#### Changes

- Updated activation strategy to align with builder strategy for more stable activation with retry logic
- Added `containerRegistryImageVersion` and `containerRegistryRepository` for improved custom image support
- Added `runAsHostUser` to fix self-hosted runner issues
- Scripts folder cleanup
- Added error/warning annotations for activation issues
- Improved serial masking
- Image environment factory addition and docker refactoring

#### Related Issues
* https://github.com/game-ci/unity-test-runner/issues/245
* https://github.com/game-ci/unity-test-runner/issues/251

#### Related PRs
- https://github.com/game-ci/unity-test-runner/pull/231 Ended up solving the same problem in this pr
- https://github.com/game-ci/documentation/pull/442

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by GabLeRoux — APPROVED** [review]
nice!

File: dist/platforms/ubuntu/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# Ensure machine ID is randomized for personal license activation
+if [[ "$UNITY_SERIAL" = F* ]]; then
+  echo "Randomizing machine ID for personal license activation"
+  dbus-uuidgen > /etc/machine-id && mkdir -p /var/lib/dbus/ && ln -sf /etc/machine-id /var/lib/dbus/machine-id
+fi
```
Comment (bogatuadrian) [inline dist/platforms/ubuntu/entrypoint.sh:7]: @AndrewKahr Hi, may I ask why this is done only for the personal license?

We are maintaining an internal fork of this repo that needed machine id randomization, but we are using Pro licenses.

**Review by davidmfinol — APPROVED** [review]

File: dist/platforms/ubuntu/entrypoint.sh (hunk 1 above)
Comment (bogatuadrian) [reply dist/platforms/ubuntu/entrypoint.sh:7]: Correct me if I'm wrong, but running say 5 test jobs in parallel with the same machine id might be problematic cause all the jobs would have the same license activation.

We randomize the machine id when running with a pro license and saw flaky activation errors disappearing. Our hypothesis is that having the same machine id leads to having the same "activation" across different parallel runs, and when one run returns the license the others can no longer use the activation they think they have.

Btw, I've written about this in the help channel on Discord as well. I'm happy to help with any other information I can.

Comment (AndrewKahr) [reply dist/platforms/ubuntu/entrypoint.sh:7]: Originally I had it randomize for all licenses but immediately on release we got a lot of bug reports for failed activations so it got reverted and likely will stay that way for the time being.

**Pull request merged**